File: Python/290. Word Pattern.py
class Solution(object):
    def wordPattern(self, pattern, s):
        """
        :type pattern: str
        :type s: str
        :rtype: bool
        """
        s=s+' '
        n = len(s)
        m = len(pattern)
        str=""
        vec=[]
        for i in range(0,n):
            if s[i]!=' ':
                str+=s[i]
            else:
                vec.append(str)
                str=""
        wordCount=len(vec)
        if m!=wordCount:
            return False
        for k in range(0,m-1):
            for l in range(k+1,m):
                if pattern[k]==pattern[l]:
                    if vec[k]==vec[l]:
                        continue
                    else:
                        return False
                elif pattern[k]!=pattern[l]:
                    if vec[k]!=vec[l]:
                        continue
                    else:
                        return False
                else:
                    return False
        return True

File: JavaScript/290. Word Pattern.js
/**
 * @param {string} pattern
 * @param {string} s
 * @return {boolean}
 */
var wordPattern = function(pattern, s) {
        s=s+' ';
        var n = s.length;
        var m = pattern.length;
        var str="";
        var vec = [];
        for(var i = 0; i<n; i++){
            if(s[i]!=' '){
                str+=s[i];
            }
            else{
                vec.push(str);
                str="";
            }
        }
        var wordCount=vec.length;
        if(m!=wordCount){return false;}
        for(var k = 0; k<m-1; k++){
            for(var l = k+1; l<m; l++){
                if(pattern[k]==pattern[l] && vec[k]==vec[l]){
                    continue;
                }
                else if(pattern[k]!=pattern[l] && vec[k]!=vec[l]){
                    continue;
                }
                else{return false;}
            }
        }
        return true;
};
